File: demo-android/app/src/main/java/com/example/demo/ui/i18n/TaskLabels.kt
package com.example.demo.ui.i18n

import androidx.annotation.StringRes
import androidx.compose.runtime.Composable
import androidx.compose.ui.res.stringResource
import com.example.demo.R
import com.example.demo.data.model.TaskPriority
import com.example.demo.data.model.TaskStatus

@StringRes
fun taskStatusStringRes(status: TaskStatus): Int = when (status) {
    TaskStatus.TODO -> R.string.status_todo
    TaskStatus.IN_PROGRESS -> R.string.status_in_progress
    TaskStatus.DONE -> R.string.status_done
}

@StringRes
fun taskPriorityStringRes(priority: TaskPriority): Int = when (priority) {
    TaskPriority.LOW -> R.string.priority_low
    TaskPriority.MEDIUM -> R.string.priority_medium
    TaskPriority.HIGH -> R.string.priority_high
}

@Composable
fun taskStatusLabel(status: TaskStatus): String {
    return stringResource(taskStatusStringRes(status))
}

@Composable
fun taskPriorityLabel(priority: TaskPriority): String {
    return stringResource(taskPriorityStringRes(priority))
}

File: demo-react-native/src/i18n/taskLabels.ts
import {TFunction} from 'i18next';

import {TaskPriority, TaskStatus} from '@/data/models/task';

export function taskStatusLabel(t: TFunction, status: TaskStatus): string {
  switch (status) {
    case TaskStatus.TODO:
      return t('statusTodo');
    case TaskStatus.IN_PROGRESS:
      return t('statusInProgress');
    case TaskStatus.DONE:
      return t('statusDone');
  }
}

export function taskPriorityLabel(t: TFunction, priority: TaskPriority): string {
  switch (priority) {
    case TaskPriority.LOW:
      return t('priorityLow');
    case TaskPriority.MEDIUM:
      return t('priorityMedium');
    case TaskPriority.HIGH:
      return t('priorityHigh');
  }
}
